src/api/app/models/bs_request_action.rb

```diff
@@ -787,18 +787,22 @@ def check_for_expand_errors!(add_revision)
       query = {}
       query[:expand] = 1 unless updatelink
       query[:rev] = source_rev if source_rev
-      dir = Xmlhash.parse(Backend::Api::Sources::Package.files(source_project, source_package, query))
 
       # Enforce revisions?
       tprj = Project.get_by_name(target_project)
       if tprj.instance_of?(Project) && tprj.find_attribute('OBS', 'EnforceRevisionsInRequests').present?
         raise ExpandError, 'updatelink option is forbidden for requests against projects with the attribute OBS:EnforceRevisionsInRequests' if updatelink
 
-        # fix the revision to the expanded sources at the time of submission
-        self.source_rev = dir['srcmd5']
+        # enforce the revision update
+        add_revision = true
+        source_rev = nil
       end
 
       if add_revision && !source_rev
+        # ensure no changes are in flight when using scripts to create and submit content
+        Backend::Api::Sources::Package.wait_service(source_project, source_package)
```

Comment:
Phew that might be a *very* long time right and be error prone right? AFAIR this part mostly runs during creation of the requests which happens in a (manual) transaction. Sounds to me like another likely source of deadlocks...

Reply:
How about doing this only for `scmsync` packages? I highly doubt that other people submit requests *before* services are finished. I'm not even sure it's possible with UI or API workflows....